app/models/mixins/purging_mixin.rb

```diff
@@ -63,11 +63,12 @@ def purge_timer
       purge_queue(*purge_mode_and_value)
     end
 
-    def purge_queue(mode, value = nil)
+    def purge_queue(mode, value = nil, callback = nil)
       MiqQueue.submit_job(
-        :class_name  => name,
-        :method_name => "purge_by_#{mode}",
-        :args        => [value]
+        :class_name   => name,
+        :method_name  => "purge_by_#{mode}",
+        :args         => [value],
+        :miq_callback => callback.present? ? callback : nil
```

Comment:
Not sure why the presence check is needed?

```suggestion
        :miq_callback => callback
```

Alternatively, if you actually need to call present on it

```suggestion
        :miq_callback => callback.presence
```

Reply:
Good point.  I originally wanted to make sure it wasn't an empty hash.  I guess we can just pass it along if it's not nil since an empty hash check isn't much of a validation anyway.